deck/views.py

```diff
@@ -31,16 +31,36 @@ class BaseEventView(object):
 class ListEvents(BaseEventView, ListView):
     template_name = 'event/event_list.html'
     queryset = Event.objects.published_ones()
+    allow_empty = True
 
     def get_context_data(self, **kwargs):
         context = super(ListEvents, self).get_context_data(**kwargs)
         if self.request.user.is_authenticated():
-            event_list = Event.objects.filter(
-                models.Q(is_published=True) |
+            event_list = context['event_list'].filter(
                 models.Q(author=self.request.user))
             context.update(event_list=event_list)
         return context
 
+    def post(self, request, *args, **kwargs):
```

Comment:
I do agree in using GET instead of POST here.

But I can’t see any problem with the search query inside the View since he’s just filtering the data here.

> So, in our case [Django framework], a “view” is the Python callback function for a particular URL, because that callback function describes which data is presented.

[Django FAQ](https://docs.djangoproject.com/en/1.10/faq/general/#django-appears-to-be-a-mvc-framework-but-you-call-the-controller-the-view-and-the-view-the-template-how-come-you-don-t-use-the-standard-names)

And about the Hexagonal Architecture, Django is an MVC/MTV framework. Despite similar in some aspects, they’re not purely comparable.

Philip Brown lists some of the drawbacks of Hexagonal Architecture:

> - A higher level of complexity
> - The cost of indirection and isolation
> - Most web applications don’t need that level of complexity

He also comments about commands:

> List users is query because you are asking a question. Commands are where you are creating, updating or deleting something.